src/content/reference/react-dom/hooks/useFormState.md

```diff
@@ -0,0 +1,219 @@
+
+If used with a server action, `useFormState` allows the server's response from submitting the form to be shown even before hydration has completed.
+
+[See more examples below.](#usage)
+
+#### Parameters {/*parameters*/}
+
+* `action`: The action to be performed when the form is submitted. It can be a server or a client action. When the action is called, it will receive the previous state of the form (initially the `initialState` that you pass, subsequently its previous return value) as its initial argument, followed by the arguments that an action normally receives.
+* `initialState`: The value you want the state to be initially. It can be any serializable value. This argument is ignored after the form is first submitted.
+
+
+#### Returns {/*returns*/}
+
+`useFormState` returns an array with exactly two values:
+
+1. The current state. During the first render, it will match the `initialState` you have passed. After the form is submitted, it will match the value returned by the action.
+2. A new action that you can pass as the `action` prop to your `form` component.
+
+---
+
+## Usage {/*usage*/}
+
+### Display a message after submitting a form {/*display-a-message-after-submitting-a-form*/}
+
+To display messages such as an error message or toast that's returned by a server action, wrap the action in a call to `useFormState`.
+
+<Sandpack>
+
+```js App.js
+import { useState } from "react";
+import { experimental_useFormState as useFormState } from "react-dom";
+import { addToCart } from "./actions.js";
+
+function AddToCartForm({itemID, itemTitle}) {
+  const [message, formAction] = useFormState(addToCart, null);
+  return (
+    <form action={formAction}>
+      <h2>{itemTitle}</h2>
+      <input type="hidden" name="itemID" value={itemID} />
+      <button type="submit">Add to Cart</button>
+      {message}
+    </form>
+  );
+}
+
+export default function App() {
+  return (
+    <>
+      <AddToCartForm itemID="1" itemTitle="Javascript: The Definitive Guide" />
+      <AddToCartForm itemID="2" itemTitle="Javascript: The Good Parts" />
+    </>
+  )
+}
+```
+
+```js actions.js
+"use server";
+
+export async function addToCart(prevState, queryData) {
+  const itemID = queryData.get('itemID');
+  if (itemID === "1") {
+    return "Added to cart";
+  } else {
+    return "Couldn't add to cart: the item is sold out.";
+  }
+}
+```
+
+```css styles.css hidden
+form {
+  border: solid 1px black;
+  margin-bottom: 24px;
+  padding: 12px
+}
+```
+
+```json package.json hidden
+{
+  "dependencies": {
+    "react": "experimental",
+    "react-dom": "experimental",
+    "react-scripts": "^5.0.0"
+  },
+  "main": "/index.js",
+  "devDependencies": {}
+}
+```
+</Sandpack>  
+
+
+### Display structured information after submitting a form {/*display-structured-information-after-submitting-a-form*/}
```

Comment:
This feels like more of a variation on the previous usage sample. Could you move this into a carousel in the previous usage example like in useState ([on react.dev](https://react.dev/reference/react/useState#examples-basic), [code](https://github.com/reactjs/react.dev/blob/main/src/content/reference/react/useState.md?plain=1#L141))